### PR TITLE
Fix bugs in SQL db accessors

### PR DESCRIPTION
**Pull request by millerdev**
- Eliminate RawQuerySetWrapper as [recommended by @snopoke](https://github.com/dimagi/commcare-hq/pull/16925#discussion_r126094790)
- Fix buggy `_order_list`

Review commits individually 🐠 and definitely wait for tests on this one.

@snopoke

### CODE REVIEW

File: corehq/form_processor/backends/sql/dbaccessors.py [hunk 1]
```diff
@@ -1187,14 +1188,19 @@ def delete_ledger_values(case_id, section_id=None, entry_id=None):
...
+    SQL does not necessarily return the rows in any particular order so
+    we need to order them ourselves.
+    
+    NOTE: this does not return the sorted list. It sorts `object_list`
+    in place using Python's built-in `list.sort`.
+    """
+    index_map = {id_: index for index, id_ in enumerate(id_list)}
+    def key(obj):
+        return index_map[getattr(obj, id_property)]
+    object_list.sort(key=key)
```
Comment (snopoke) [inline corehq/form_processor/backends/sql/dbaccessors.py]: nice, this is much better.

**Review by snopoke — APPROVED** [review]

**Comment by millerdev** [comment]
hmm, the `[ci skip]` directive seems to have created a confusing "All checks have passed"  message even though the tests are still running. I'm watching the build at https://travis-ci.org/dimagi/commcare-hq/builds/251242733 and will merge when that passes.

**Comment by snopoke** [comment]
👍

**Comment by millerdev** [comment]
✅

**Pull request merged**